### PR TITLE
Update to hostap module

### PR DESCRIPTION
**Pull request by jukkar**
This PR takes changes from Nordic downstream hostap glue code and upstreams them.
Sorry about the amount of commits in review, it is somewhat unavoidable.

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]
Not sure if we need those `cherry picked from commit xxx` lines, those commits are in entirely different repository, not even a fork, so could be somewhat confusing. But I hav eno strong opinion on this.

**Comment by jukkar** [comment]
> Not sure if we need those `cherry picked from commit xxx` lines, those commits are in entirely different repository, not even a fork, so could be somewhat confusing.

I was also thinking the same, but left them at least for the initial review. The "cherry-picked from..." string marks those commits I needed to resolve conflicts. If the string is missing, the commit applied as is. This PR needs still some work to to fix the CI so I can remove them but leave my signed-off-by to mark the same thing.

**Comment by jukkar** [comment]
Changes:
* rebased on top of latest upstream
* removed "cherry picked" lines from commits
* fixed compliance check issues

**Comment by jukkar** [comment]
@krish2718 @dleach02 @rlubos this is ready for a review after the updated hostap module is merged, please take a look.

File: modules/hostap/Kconfig [hunk 1]
```diff
@@ -59,29 +61,60 @@ config WIFI_NM_WPA_SUPPLICANT_DEBUG_LEVEL
...
+	select MBEDTLS_ECP_C
+	select MBEDTLS_ECP_ALL_ENABLED
+	select MBEDTLS_PKCS5_C
+	select MBEDTLS_MAC_CMAC_ENABLED
+#	select MBEDTLS_CIPHER_MODE_CBC
+#	select MBEDTLS_CIPHER_MODE_CTR
+#	select MBEDTLS_LEGACY_CRYPTO_C
+#	select MBEDTLS_CTR_DRBG_C
+#	select MBEDTLS_PK_WRITE_C
+#	select MBEDTLS_KEY_EXCHANGE_ALL_ENABLED
```
Comment (rlubos) [inline modules/hostap/Kconfig:117]: Any reason to have those commented-out entries?

**Review by rlubos — APPROVED** [review]

File: modules/hostap/Kconfig (hunk 1 above)
Comment (jukkar) [reply modules/hostap/Kconfig:117]: The mbedtls support needs more work so some of the entries were commented out. I can certainly remove those too, and keep track of them separately.

**Review by krish2718 — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**